dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -196,12 +197,18 @@ func startSyncWriter(queue util.Queue) util.Queue {
 }
 
 func (cw *ClientWriter) sendSuccessPiece(piece *Piece, cost time.Duration) {
-	cw.api.ReportPiece(piece.SuperNode, &types.ReportPieceRequest{
+	resp, e := cw.api.ReportPiece(piece.SuperNode, &types.ReportPieceRequest{
 		TaskID:     piece.TaskID,
 		Cid:        cw.cfg.RV.Cid,
 		DstCid:     piece.DstCid,
 		PieceRange: piece.Range,
 	})
+	if e != nil {
+		logrus.Errorf("sendSuccessPiece error:%v", e)
+	}
+	if resp.Code != constants.CodeGetPieceReport {
```

Comment:
I think here should be in an `else` block. 

```
if e := nil {
    logrus.Errorf("failed to report piece: %v", err)
} else if resp.Code != constants.CodeGetPieceReport {
    logrus.Errorf("send success piece report to supernode failed")
}
```

Reply:
Done